### PR TITLE
[FLINK-30757] Upgrade busybox version to a pinned version for operator

### PR DESCRIPTION
**Pull request by xshipeng**
## What is the purpose of the change

Since [e2e test](https://github.com/apache/flink-kubernetes-operator/actions/runs/3895555690/jobs/6662337363) is flaky with latest busybox image in init container, we [pinned](https://github.com/apache/flink-kubernetes-operator/commit/8e41ed5ec1adda9ae06bc0b6203abf42939fbf2b) the image version to a relatively old version. This pull request would like to enable e2e test print init container log and try to upgrade busybox version to the latest pinned version.

## Brief change log
  - Add log printing for init container if e2e test is failed.
  - Upgrade the busybox version to the latest in a pinned way.

## Verifying this change

This change is already covered by existing tests, such as `e2e-tests/test_application_kubernetes_ha.sh`.

To reproduce the failed tests locally and test modified init container log printing code:
  - Change the init container command in `e2e-tests/data/flinkdep-cr.yaml` to `command: ['/bin/sh', '-c', 'wget https://repo1.maven.org/maven2/org/apache/flink/flink-examples-streaming_2.12/1.14.4/flink-examples-streaming_2.12-1.14.4.jar -O /flink-artifact/myjob.jar; echo "test error msg"; exit 1;']`.
  - Run `./e2e-tests/test_application_kubernetes_ha.sh`.
  - Part of the print log will be 
  ```
...
Flink logs:
Printing init container logs
--------BEGIN CURRENT LOG for flink-example-statemachine-599c95bbf-42wtl:artifacts-fetcher--------
Connecting to repo1.maven.org (198.18.3.208:443)
wget: note: TLS certificate validation not implemented
saving to '/flink-artifact/myjob.jar'
myjob.jar            100% |********************************|  267k  0:00:00 ETA
'/flink-artifact/myjob.jar' saved
test error msg
--------END CURRENT LOG--------
--------BEGIN PREVIOUS LOGS for flink-example-statemachine-599c95bbf-42wtl:artifacts-fetcher--------
Connecting to repo1.maven.org (198.18.3.208:443)
wget: note: TLS certificate validation not implemented
saving to '/flink-artifact/myjob.jar'
myjob.jar            100% |********************************|  267k  0:00:00 ETA
'/flink-artifact/myjob.jar' saved
test error msg
--------END PREVIOUS LOGS--------
Printing main container logs
--------BEGIN CURRENT LOG for flink-example-statemachine-599c95bbf-42wtl:flink-main-container--------
--------END CURRENT LOG--------
Printing init container logs
Printing main container logs
--------BEGIN CURRENT LOG for flink-kubernetes-operator-9744c66bd-27q2r:flink-kubernetes-operator--------
Starting Operator
...
```

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency):  no
  - The public API, i.e., is any changes to the `CustomResourceDescriptors`: no
  - Core observer or reconciler logic that is regularly executed: no

## Documentation

  - Does this pull request introduce a new feature?  no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by xshipeng** [comment]
It looks like busybox latest (1.36.0) is unstable -> https://github.com/docker-library/busybox/issues/162

**Comment by xshipeng** [comment]
Hi @gyfora @gaborgsomogyi , the pipeline checks with 1.35.0 busybox version passed. Should we wait for https://github.com/docker-library/busybox/issues/162 to be solved or we are good with 1.35.0 version?

**Comment by gaborgsomogyi** [comment]
> Hi @gyfora @gaborgsomogyi , the pipeline checks with 1.35.0 busybox version passed. Should we wait for https://github.com/docker-library/busybox/issues/162 to be solved or we are good with 1.35.0 version?

I'm fine with 1.35.0 as long as it's stable. The main feature what I was personally missing is the failure log for init containers.

File: e2e-tests/utils.sh [hunk 1]
```diff
@@ -175,20 +175,33 @@ function debug_and_show_logs {
...
-          echo "Current logs for $pod:$container: "
-          kubectl logs $pod $container;
-          restart_count=$(kubectl get pod $pod -o jsonpath='{.status.containerStatuses['$i'].restartCount}')
-          if [[ ${restart_count} -gt 0 ]];then
-            echo "Previous logs for $pod: "
-            kubectl logs $pod $container --previous
-          fi
-        done
+        print_pod_container_logs "$pod" "{.spec.initContainers[*].name}" "{.status.initContainerStatuses[*].restartCount}"
+        print_pod_container_logs "$pod" "{.spec.containers[*].name}" "{.status.containerStatuses[*].restartCount}"
```
Comment (gaborgsomogyi) [inline e2e-tests/utils.sh:179]: I would add `echo "Printing init container logs"` and `echo "Printing main container logs"` for better understanding.

File: e2e-tests/utils.sh [hunk 2]
```diff
@@ -175,20 +175,33 @@ function debug_and_show_logs {
...
+
+  containers=(`kubectl get pods $pod -o jsonpath=$container_path`)
+  restart_counts=(`kubectl get pod $pod -o jsonpath=$restart_count_path`)
+
+  if [[ -z "$containers" ]];then
+    return 0
+  fi
+
+  for idx in "${!containers[@]}"; do
+    echo "Current logs for $pod:${containers[idx]}: "
```
Comment (gaborgsomogyi) [inline e2e-tests/utils.sh:196]: I know it was implemented this way so it's a "new feature" but can we add some begin/end marker for the logs?
Maybe
```
--------BEGIN CURRENT LOGS for flink-example-statemachine-dddb4d664-m6n5b:flink-main-container--------
foo
--------END CURRENT LOGS--------
```
or something. My eye could parse it easier.

**Review by gaborgsomogyi — COMMENTED** [review]
Thanks for the contribution, decent job. I've tried it out and looks good, only added some comments to make the output easier to read.

**Review by gaborgsomogyi — APPROVED** [review]
LGTM.

**Comment by gaborgsomogyi** [comment]
cc @gyfora @mbalassi

**Comment by xshipeng** [comment]
Hi @gyfora, the latest Java integration test failed. Could you please help me retrigger the ci pipeline again? I tested the integration test locally and it passed.

**Pull request merged**